include/AutomatableModel.h

```diff
@@ -80,6 +82,32 @@ class LMMS_EXPORT AutomatableModel : public Model, public JournallingObject
 
 	virtual ~AutomatableModel();
 
+	// Implement those by using the MODEL_IS_VISITABLE macro
+	virtual void accept(ModelVisitor& v) = 0;
+	virtual void accept(ConstModelVisitor& v) const = 0;
+	// use this to make subclasses visitable
+#define MODEL_IS_VISITABLE \
+	void accept(ModelVisitor& v) override { v.visit(*this); } \
+	void accept(ConstModelVisitor& v) const override { v.visit(*this); }
+
+public:
+	//! Return this class casted to Target, or nullptr if impossible
+	template<class Target>
+	Target* dcast(bool doThrow = false)
```

Comment:
Could we have a more descriptive function name?

Reply:
It could be renamed to `dynamic_cast`, I think that explains it best. Any other ideas/preferences?

Reply:
`dynamic_cast` itself is a C++ keyword. We can use something like `dynamicCast` if desired. Note that there's also `ModelView::castModel` which uses `dynamic_cast`.